lib/spack/spack/build_systems/python.py

```diff
@@ -128,191 +112,79 @@ def import_modules(self):
 
         modules = [mod for mod in modules if re.match('[a-zA-Z0-9._]+$', mod)]
 
         tty.debug('Detected the following modules: {0}'.format(modules))
 
         return modules
 
-    def setup_file(self):
-        """Returns the name of the setup file to use."""
-        return 'setup.py'
-
     @property
     def build_directory(self):
-        """The directory containing the ``setup.py`` file."""
-        return self.stage.source_path
-
-    def python(self, *args, **kwargs):
-        inspect.getmodule(self).python(*args, **kwargs)
-
-    def setup_py(self, *args, **kwargs):
-        setup = self.setup_file()
-
-        with working_dir(self.build_directory):
-            self.python('-s', setup, '--no-user-cfg', *args, **kwargs)
-
-    # The following phases and their descriptions come from:
-    #   $ python setup.py --help-commands
-
-    # Standard commands
+        """The root directory of the Python package.
 
-    def build(self, spec, prefix):
-        """Build everything needed to install."""
-        args = self.build_args(spec, prefix)
+        This is usually the directory containing one of the following files:
 
-        self.setup_py('build', *args)
-
-    def build_args(self, spec, prefix):
-        """Arguments to pass to build."""
-        return []
-
-    def build_py(self, spec, prefix):
-        '''"Build" pure Python modules (copy to build directory).'''
-        args = self.build_py_args(spec, prefix)
-
-        self.setup_py('build_py', *args)
-
-    def build_py_args(self, spec, prefix):
-        """Arguments to pass to build_py."""
-        return []
-
-    def build_ext(self, spec, prefix):
-        """Build C/C++ extensions (compile/link to build directory)."""
-        args = self.build_ext_args(spec, prefix)
-
-        self.setup_py('build_ext', *args)
-
-    def build_ext_args(self, spec, prefix):
-        """Arguments to pass to build_ext."""
-        return []
-
-    def build_clib(self, spec, prefix):
-        """Build C/C++ libraries used by Python extensions."""
-        args = self.build_clib_args(spec, prefix)
-
-        self.setup_py('build_clib', *args)
+        * ``pyproject.toml``
+        * ``setup.cfg``
+        * ``setup.py``
+        """
+        return self.stage.source_path
 
-    def build_clib_args(self, spec, prefix):
-        """Arguments to pass to build_clib."""
+    def install_options(self, spec, prefix):
+        """Extra arguments to be supplied to the setup.py install command."""
         return []
 
-    def build_scripts(self, spec, prefix):
-        '''"Build" scripts (copy and fixup #! line).'''
-        args = self.build_scripts_args(spec, prefix)
-
-        self.setup_py('build_scripts', *args)
-
-    def build_scripts_args(self, spec, prefix):
-        """Arguments to pass to build_scripts."""
+    def global_options(self, spec, prefix):
+        """Extra global options to be supplied to the setup.py call before the install
```

Comment:
Still `setup.py` here as well.

Reply:
Yep, when setuptools is in use, `pip` seems to run the `setup.py` script itself, and provides this option to inject necessary build args.